tests/system_tests/wandb_backend_spy/proxy.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import threading
+import time
+import traceback
+from typing import Iterator
+
+import fastapi
+import httpx
+import uvicorn
+
+from .spy import WandbBackendSpy
+
+
+@contextlib.contextmanager
+def spy_proxy(
+    proxy_port: int,
+    target_host: str,
+    target_port: int,
+) -> Iterator[WandbBackendProxy]:
+    """A context manager that proxies a W&B backend.
+
+    The proxy server will be available at 127.0.0.1 on the given port.
+    The server is shut down when the context manager exits.
+
+    Args:
+        proxy_port: The port on which to run the proxy server.
+        target_host: The hostname of the W&B backend to proxy.
+        target_port: The port of the W&B backend to proxy.
+    """
```

Comment:
should we add the Returns, and Raises fields in the docs?

Reply:
added a Yields section

there shouldn't be a Raises section because exceptions aren't part of the contract---this function is expected to always succeed, and exceptions should be treated as programming errors